docs/practice-playbook.md

```diff
@@ -199,49 +288,149 @@ Reviewers should _always_ feel free to leave comments expressing that something
+Regardless of MR/PR or commiting directly to main/master, testing needs to be completed on new functionality before it is committed and pushed to main/master.
+`#YouBreakItYouBuyIt`
 
-_This section was derived, with modifications, from [Google Engineering Practices Documentation](https://github.com/google/eng-practices)_
+#### HOOKS
+We strongly encourage the use of commit hooks to further ensure code quality. These hooks can range from enforcing commit formats to running unit tests and may be left up to the team to decide.
+
+#### MERGE REQUEST COMMENTS
+We encourage comments/suggestions/questions/discussion/etc. on MR per our belief in strong opinions loosely held >> better resulting code
+
+#### REBASE
+We encourage squashing and rebasing to preserve the cleanliness and readability of the git history on the master branch. This should only be performed by an engineer that understands the rebasing process in order to avoid causing irreparable damage to the master branch. If done correctly, there should be no explicit merge commits.
+
+NOTE: Certain technologies (i.e. GitLab) default behavior creates merge commits. This can be changed.
+
+#### COMMIT MESSAGES
+Commit messages should be a brief, concise description in imperative tense of what the commit adds, with the appropriate authors (alternating authors or using tools such as git with .git-together), and the ID of the corresponding story. Using industry standards such as [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) are not required but teams may choose to follow any given industry standard.
+
+#### BRANCH NAMING
+Branch naming should be clear and concise. We recommend the convention of including the story ID followed by a few words for the branch's purpose, using dashes (-) as the delimiter.
+
+#### BRANCH MAINTENANCE
+Cleanup/remove branches post merge completion. Developers should be wary and not have too many inactive/stale branches linger. Abandoned branches should be removed to avoid Git pollution.
+
+
+
+### CI/CD Pipeline
+
+**Continuous Integration**
+
+We beleive CI begins at the local development env. This includes the tools to run automated tests, linting, and other checks on branches **BEFORE** you merge up.
+
+
+
+We believe CI is non-negotiable and must begin at the initial conception of development to ensure comprehensive software security, testing, and fast feedback on the main branch health. Furthermore it empowers the ability of the team to hold to agile practices.  
+
+CI stages should include at a minimum
+- linting
+- unit tests
+- static code analysis
+- dependency scans
+
+
+**Continuous Delivery**
+We believe that any merge to main should be able to deploy to production and main should always be deployable. Merges should be self-contained and not dependant upon another branch.
```

Comment:
TODO: Need to address this more. Maybe remove for now?

Reply:
good engough for now. I added a comment to re-attack